### ISSUE TITLE
Publishing with JDK17 fails due to sbt-osgi

### ISSUE THREAD

**Issue by pvlugter** [akka/akka#31132]
The JDK 17 builds are failing with `ConcurrentModificationException`s for the `osgiBundle` task.

https://github.com/akka/akka/runs/5225034561?check_suite_focus=true

https://github.com/akka/akka/runs/5225034721?check_suite_focus=true

```
[error] java.util.ConcurrentModificationException
[error] 	at java.base/java.util.TreeMap.callMappingFunctionWithCheck(TreeMap.java:750)
[error] 	at java.base/java.util.TreeMap.computeIfAbsent(TreeMap.java:604)
[error] 	at aQute.bnd.osgi.Jar.putResource(Jar.java:259)
[error] 	at aQute.bnd.osgi.Jar$1.visitFile(Jar.java:186)
[error] 	at aQute.bnd.osgi.Jar$1.visitFile(Jar.java:167)
[error] 	at java.base/java.nio.file.Files.walkFileTree(Files.java:2811)
[error] 	at aQute.bnd.osgi.Jar.buildFromDirectory(Jar.java:166)
[error] 	at aQute.bnd.osgi.Jar.<init>(Jar.java:109)
[error] 	at aQute.bnd.osgi.Jar.<init>(Jar.java:138)
[error] 	at aQute.bnd.osgi.Analyzer.setClasspath(Analyzer.java:1465)
[error] 	at com.typesafe.sbt.osgi.Osgi$.bundleTask(Osgi.scala:57)
[error] 	at com.typesafe.sbt.osgi.SbtOsgi$.$anonfun$defaultOsgiSettings$1(SbtOsgi.scala:55)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] 	at sbt.Execute.work(Execute.scala:291)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[error] 	at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by patriknw** [comment]
we could drop this test completely, or just for JDK 17. osgi is not a priority, #28304

**Comment by patriknw** [comment]
ah, it's not a test but the actual build

**Comment by pvlugter** [comment]
Yeah, this is the local publish step, which it's getting to after tests passing. Could not run the publish check for JDK 17?

**Comment by patriknw** [comment]
absolutely, created PR https://github.com/akka/akka/pull/31135 for that

**Comment by patriknw** [comment]
The issue is not really fixed, but we don't need to publish with JDK 17 (yet)

**Comment by raboof** [comment]
(since https://github.com/akka/akka/blob/main/.github/workflows/nightly-builds.yml#L206 points here, let's keep this issue open for eventually supporting publishing with jdk17)

**Comment by raboof** [comment]
this problem possibly has been fixed already between sbt-osgi 0.9.4 and 0.9.6, but we can't update yet.

**Comment by patriknw** [comment]
Obsolete, see https://github.com/akka/akka/issues/28304
